user-dev-workaround.sh

```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+set -uxe
+
+# Workaround because MediaGoblin doesn't fully respect data_basedir in the
+# mediagoblin.ini.
+#
+# When data_basedir is defined in mediagoblin.ini, MediaGoblin writes uploaded
+# files to that location, but when it tries to serve files, it still looks for
+# them in ./user_dev/media/public.
+#
+# As a workaround, create a symlink from the incorrect path to the correct
+# path.
+
+CORRECT_SERVING_DIR="${MEDIAGOBLIN_HOME_DIR}/media/public"
+INCORRECT_SERVING_DIR="./user_dev/media/public"
+
+mkdir --parents "$CORRECT_SERVING_DIR"
+mkdir --parents $(dirname "$INCORRECT_SERVING_DIR")
+ln -s "$CORRECT_SERVING_DIR" "$INCORRECT_SERVING_DIR"
```

Comment:
I believe this should read:
```shell
ln -sf "$CORRECT_SERVING_DIR" "$INCORRECT_SERVING_DIR"
```

In my case, the container starts the first time, then on subsequent runs it fails and dies with an error that this path already exists.

This is the full log out. It's always truncated like that, not sure why. Other containers do not do the same thing.
```
No host-defined mediagoblin.ini file, using default'
defined mediagoblin.ini file, using default
--no-dereference --recursive mediagoblin:mediagoblin /var/lib/mediagoblin
iagoblin -c ./user-dev-workaround.sh
T_SERVING_DIR=/var/lib/mediagoblin/media/public
ECT_SERVING_DIR=./user_dev/media/public
--parents /var/lib/mediagoblin/media/public
me ./user_dev/media/public
--parents ./user_dev/media
/var/lib/mediagoblin/media/public ./user_dev/media/public
ed to create symbolic link './user_dev/media/public/public': File exists
mediagoblin_host.ini ']'
No host-defined mediagoblin.ini file, using default'
defined mediagoblin.ini file, using default
--no-dereference --recursive mediagoblin:mediagoblin /var/lib/mediagoblin
iagoblin -c ./user-dev-workaround.sh
T_SERVING_DIR=/var/lib/mediagoblin/media/public
ECT_SERVING_DIR=./user_dev/media/public
--parents /var/lib/mediagoblin/media/public
me ./user_dev/media/public
--parents ./user_dev/media
/var/lib/mediagoblin/media/public ./user_dev/media/public
ed to create symbolic link './user_dev/media/public/public': File exists
```

Reply:
Can you file a separate bug that includes the full commands you're running to hit this? I'm not able to reproduce this.

Reply:
I'm not real sure how I keep hitting this, but I keep hitting it. I'm using unRAID so I'm not issuing commands directly. This is the command that starts the container: 
```shell
/usr/local/emhttp/plugins/dynamix.docker.manager/scripts/docker run -d --name='mediagoblin' --net='my-bridge' -e TZ="America/Los_Angeles" -e HOST_OS="Unraid" -p '6543:6543/tcp' -v '/mnt/user/MediaGoblin':'/var/lib/mediagoblin':'rw' -v '/mnt/user/MediaGoblin/mediagoblin_host.ini':'/srv/mediagoblin.example.org/mediagoblin/mediagoblin_host.ini':'rw' --tty --detach 'mtlynch/mediagoblin:latest' ```

The ini file I'm passing in is default + video and audio plugins. If I figure it out I'll open an issue. For now, I just modified the file in the container so it would start without errors.